include/ifc/pathname.hxx

```diff
@@ -103,8 +104,9 @@ namespace ifc {
 
         value_type back() const
         {
-            IFCASSERT(length() > 0);
-            return begin()[length() - 1];
+            std::size_t length = this->length();
```

Comment:
Let's use a local variable that isn't also the name of the (base) class function.
I would have used `n`, but I can understand someone wanting a longer name - but, please,  not `length` :-)